src/typescript_proto_library.bzl

```
@@ -122,6 +168,9 @@ def typescript_proto_library_aspect_(target, ctx):
     [js_outputs, js_outputs_es6, dts_outputs] = _get_outputs(target, ctx)
     protoc_outputs = dts_outputs + js_outputs + js_outputs_es6
 
+    if ctx.attr.mode == "grpc-js" and ctx.attr.generate != "grpc-node":
```

Nit: Let's put the checks at the beginning of the method so they fail fast